packages/web/integrations/tfm/router.ts

```diff
@@ -0,0 +1,71 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+import {
+  SplitTokenInQuote,
+  Token,
+  TokenOutGivenInRouter,
+} from "@osmosis-labs/pools";
+
+import { GetSwapRouteResponse, GetTokensResponse } from "./types";
+
+export class TfmRemoteRouter implements TokenOutGivenInRouter {
+  protected readonly baseUrl: URL;
+
+  constructor(
+    protected readonly osmosisChainId: string,
+    protected readonly tfmBaseUrl: string
+  ) {
+    this.baseUrl = new URL(tfmBaseUrl);
+  }
+
+  async getRoutableCurrencyDenoms(): Promise<string[]> {
+    const params = new URLSearchParams();
+    params.append("isTrading", "True");
+    const queryUrl = new URL(
+      "/api/v1/ibc/chain/osmosis-1/tokens",
+      this.baseUrl.toString()
+    );
+    const response = await fetch(queryUrl.toString());
+    const result = (await response.json()) as GetTokensResponse;
```

Comment:
We can use the apiClient and simplify this further
```suggestion
    const result = await apiClient<GetTokensResponse>(queryUrl.toString());
```